bindings/pybind11/idyntree_vector_casters.h

```diff
@@ -0,0 +1,277 @@
+ * has is an iDynTree::VectorFixSize. This specific implementation is used
+ * when the the object is not an iDynTree::VectorFixSize
+ */
+template<typename>
+struct is_vector_fixsize : std::false_type {};
+
+/**
+ * is_vector_fixsize is used to build a type-dependent expression that check if an element
+ * has is an iDynTree::VectorFixSize. This specific implementation is used
+ * when the the object not an iDynTree::VectorFixSize
+ */
+template<unsigned int N>
+struct is_vector_fixsize<::iDynTree::VectorFixSize<N>> : std::true_type {};
+
+/**
+ * is_matrix_fixsize is used to build a type-dependent expression that check if an element
+ * has is an iDynTree::MatrixFixSize. This specific implementation is used
+ * when the the object is not an iDynTree::MatrixFixSize
+ */
+template<typename>
+struct is_matrix_fixsize : std::false_type {};
+
+/**
+ * is_matrix_fixsize is used to build a type-dependent expression that check if an element
+ * has is an iDynTree::MatrixFixSize. This specific implementation is used
+ * when the the object is an iDynTree::MatrixFixSize
+ */
+template<unsigned int nRows, unsigned int nCols>
+struct is_matrix_fixsize<::iDynTree::MatrixFixSize<nRows, nCols>> : std::true_type {};
+
+/**
+ * vector_size_at_compile_time is a utility metafunction to get the compile-time size of an iDynTree
+ * vector. This specific implementation is used with VectorDynSize.
+ */
+template<typename>
+struct vector_size_at_compile_time
+{
+    constexpr static int value = -1;
+};
+
+/**
+ * vector_size_at_compile_time is a utility metafunction to get the compile-time size of an iDynTree
+ * vector. This specific implementation is used with VectorFixSize.
+ */
+template<unsigned int N>
+struct vector_size_at_compile_time<::iDynTree::VectorFixSize<N>>
+{
+    constexpr static int value = N;
+};
+
+/**
+ * matrix_size_at_compile_time is a utility metafunction to get the compile-time rows and cols of
+ * an iDynTree matrix. This specific implementation is used with MatrixDynSize.
+ */
+template<typename>
+struct matrix_size_at_compile_time
+{
+    constexpr static int rows = -1;
+    constexpr static int cols = -1;
+};
+
+/**
+ * matrix_size_at_compile_time is a utility metafunction to get the compile-time rows and cols of
+ * an iDynTree matrix. This specific implementation is used with MatrixFixSize.
+ */
+template<unsigned int nRows, unsigned int nCols>
+struct matrix_size_at_compile_time<::iDynTree::MatrixFixSize<nRows, nCols>>
+{
+    constexpr static int rows = nRows;
+    constexpr static int cols = nCols;
+};
+
+
+
+/**
+ * type_caster implementation for the iDynTree Vectors (both fixed and dynamical size)
+ */
+template <typename Type>
+struct type_caster<Type,
+                   enable_if_t<is_vector_fixsize<Type>::value
+                               || std::is_same<Type, iDynTree::VectorDynSize>::value>>
+{
+private:
+
+    template <typename U>
+    static bool checkSize(const std::size_t size, const U& value)
+    {
```

Comment:
Can you please put the opening `{` on the same line as the function definition?

Reply:
7552d12ff13be2d1e75c8b8e3484e5b265d821d5